File: Go/find-and-replace.go
package main

import (
	"log"
	"os"
	"strings"
)

var (
	oldString string
	newString string
	filePath  string
)

func main() {
	content, err := readAFile(filePath)
	if err != nil {
		log.Fatal(err)
	}
	newContents := strings.Replace(string(content), (oldString), (newString), -1)
	err = writeContnetToFile([]byte(filePath), newContents)
	if err != nil {
		log.Fatal(err)
	}
}

// Don't append and write to file
func writeContnetToFile(filepath []byte, content string) error {
	err := os.WriteFile(content, filepath, 0644)
	if err != nil {
		return err
	}
	return nil
}

// Check if a file exists
func fileExists(filename string) bool {
	info, err := os.Stat(filename)
	if err != nil {
		return false
	}
	return !info.IsDir()
}

// Read a file and return the contents
func readAFile(path string) ([]byte, error) {
	content, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	return content, nil
}

File: JavaScript/find-and-replace.js
function main() {
    var text = "This is a string with a few words in it.";
    console.log(findAndReplace(text, "A", "This"));
}

main()

// find and replace
function findAndReplace(oldString, newString, searchString) {
    var newString = newString.replace(/\$/g, "\\$");
    var regExp = new RegExp(searchString, "g");
    return oldString.replace(regExp, newString);
}